src/crawlee/sessions/_session.py

```diff
@@ -193,17 +193,22 @@ def is_blocked_status_code(
         self,
         *,
         status_code: int,
-        additional_blocked_status_codes: list[int] | None = None,
+        additional_blocked_status_codes: set[int] | None = None,
+        ignore_http_error_status_codes: set[int] | None = None,
```

Comment:
Is there even a good reason for this method to be on the `Session` class? Maybe we could move it to the `BasicCrawler` instead.

Reply:
I'm not sure if this is a compelling reason. But if our public API allows configuring errors at the Session level through:

```python
Session(blocked_status_codes=[...])
```

and

```python
SessionPool(create_session_settings={'blocked_status_codes': [...]})
```

then I believe it makes sense to have this method at the Session class level.

Reply:
Huh, yeah, I'm not sure if enabling this in `SessionPool` is in fact necessary, but let's not change it in this PR :smile: